cmd/downloadHistory.go

```diff
@@ -0,0 +1,253 @@
+  blazectl download-history --server http://localhost:8080/fhir Patient DFRE25Q627JVEWOS > patient-history.ndjson
+  blazectl download-history --server http://localhost:8080/fhir Patient > patients-history.ndjson
+  blazectl download-history --server http://localhost:8080/fhir > system-history.ndjson`,
+	ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		return resourceTypes, cobra.ShellCompDirectiveNoFileComp
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		err := createClient()
+		if err != nil {
+			return err
+		}
+		var stats commandStats
+		startTime := time.Now()
+
+		var file *os.File
+		if outputFile == "" {
+			file = os.Stdout
+		} else {
+			file = createOutputFileOrDie(outputFile)
+		}
+		sink := bufio.NewWriter(file)
+		defer file.Close()
+		defer file.Sync()
+		defer sink.Flush()
+
+		bundleChannel := make(chan downloadBundle, 2)
+
+		var resourceType string
+		var resourceId string
+		if len(args) > 0 {
+			resourceType = args[0]
+		} else {
+			resourceType = ""
+			resourceId = ""
+		}
+		if len(args) > 1 {
+			resourceId = args[1]
+		}
+
+		go downloadHistory(client, resourceType, resourceId, bundleChannel)
+
+		for bundle := range bundleChannel {
+			stats.totalPages++
+
+			if bundle.err != nil || bundle.errResponse != nil {
+				fmt.Printf("Failed to download resources: %v\n", bundle.err)
+
+				stats.error = bundle.errResponse
+				stats.totalDuration = time.Since(startTime)
+				fmt.Println(stats.String())
+				os.Exit(1)
+			} else {
+				stats.requestDurations = append(stats.requestDurations, bundle.stats.requestDuration)
+				stats.processingDurations = append(stats.processingDurations, bundle.stats.processingDuration)
+				stats.totalBytesIn += bundle.stats.totalBytesIn
+
+				resources, inlineOutcomes, err := writeResources(bundle.responseBody, sink)
+				stats.resourcesPerPage = append(stats.resourcesPerPage, resources)
+				stats.inlineOperationOutcomes = append(stats.inlineOperationOutcomes, inlineOutcomes...)
+
+				if err != nil {
+					fmt.Printf("Failed to write downloaded resources received from request to URL %s: %v\n", bundle.associatedRequestURL.String(), err)
+					os.Exit(2)
+				}
+			}
+		}
+
+		stats.totalDuration = time.Since(startTime)
+		fmt.Fprint(os.Stderr, stats.String())
+		return nil
+	},
+}
+
+// downloadResources tries to download all resources of a given resource type from a FHIR server using
+// the given client. Resources that are downloaded can optionally be limited by a given FHIR search query.
+// The download respects pagination, i.e. it follows pagination links until there is no other next link.
+//
+// Downloaded resources as well as errors are sent to a given result channel.
+// As soon as an error occurs, it is written to the channel and the channel and closed thereafter.
+func downloadHistory(client *fhir.Client, resourceType string, resourceId string, resChannel chan<- downloadBundle) {
```

Comment:
Should be a function called `ExpandPages(client, initialRequest, resChannel)` in the client file